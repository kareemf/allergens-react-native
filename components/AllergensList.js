import React from 'react';
import {
  Text,
  View,
  FlatList
} from 'react-native';

//TODO: propTypes
export default class AllergensList extends React.Component {
  render() {
    const { data } = this.props;

    return (
      <View>
        <FlatList
          data={data}
          keyExtractor={(item, index) => item.name}
          renderItem={({item}) => <Text>{item.name}</Text>}
        />
      </View>
    );
  }
}
