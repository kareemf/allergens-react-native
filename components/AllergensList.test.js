import React from 'react';
import AllergensList from './AllergensList';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create( < AllergensList / > ).toJSON();
  expect(rendered).toBeTruthy();
});

it('renders correctly', () => {
  // given jest snapshot data
  const snapshotData = [{
    name: 'Foo'
  }];

  // when component is rendered
  const rendered = renderer.create( <AllergensList data={snapshotData} /> ).toJSON();

  // then rendered tree matches snapshot
  expect(rendered).toMatchSnapshot();
});

