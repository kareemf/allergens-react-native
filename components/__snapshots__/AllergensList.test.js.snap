// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "name": "Foo",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Foo
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
